### ISSUE TITLE
[Bug]: Garrison loadout from save fails, preventing mission from starting

### ISSUE THREAD

**Issue by Thadah** [igorkis-scrts/A3-Antistasi-Plus#56]
### Describe the bug

Hi, we tried loading a Tanoa mission after playing normally for a couple of weeks and after the scripts initialize and players join the server world we are stuck in the "Waiting for players and server init" black screen forever. What's causing this is the garrisons not loading properly.

```
Error in expression <son = +(_x select 1);
                        {

                                if (_x find "loadouts_rebel" != 0) then { co>
Error position: <_x find "loadouts_rebel" != 0) then { co>
Error Undefined variable in expression: _x
```

The save was done automatically after the last player disconnected. After that the server was left idling for several hours before restarting it.

### How to reproduce

Just by making one of the many persistent saves of the game and trying to reload it after a server restart.

### Version

1.7.3

### Have you altered the code?

No

### Map

Tanoa

### What server?

Private dedicated server with Headless client(s)

### Time bug occured (Server time/UTC)

13:00 (UTC)

### Mods

```html
3CB Factions 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=1673456286
ace 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=463939057
ACE 3 Extension (Animations and Actions) 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=766491311
ACE 3 Extension (Placeables) 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=866772689
ACE Compat - RHS Armed Forces of the Russian Federation 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=773131200
ACE Compat - RHS United States Armed Forces 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=773125288
ACE Compat - RHS: GREF 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=884966711
ACE Compat - RHS: SAF 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=2174495332
ACEX 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=708250744
ACRE2 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=751965892
BackpackOnChest - Redux 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=2372036642
CBA_A3 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=450814997
Enhanced Movement 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=333310405
Grenades In Tank Hatches for ACE 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=2398240266
Immersion Cigs 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=753946944
KKA3 Injured Animations 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=2172423762
RHSAFRF 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=843425103
RHSGREF 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=843593391
RHSSAF 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=843632231
RHSUSAF 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=843577117
UnderSiege Patches & Insignias 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=1113616706
VET_Unflipping 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=1703187116
WMO - Walkable Moving Objects 	Steam 	http://steamcommunity.com/sharedfiles/filedetails/?id=925018569
```

### Additional context

Snippet of the logs when the server failed

```
13:20:02 "6712.77: [Antistasi] | INFO | loadServer Starting."
13:20:02 "6712.77: [Antistasi] | INFO | Starting Persistent Load."
13:20:02 Weather was forced to change
13:20:02 Error in expression <son = +(_x select 1);
                        {

                                if (_x find "loadouts_rebel" != 0) then { co>
13:20:02   Error position: <_x find "loadouts_rebel" != 0) then { co>
13:20:02   Error Undefined variable in expression: _x
13:20:02 File /documents/A3-Antistasi-Plus/A3-Antistasi/functions/Save/fn_loadStat.sqf..., line 151
13:20:02 2021-08-17 13:19:21:464 | Antistasi | Info | File: initFuncs.sqf | initFuncs started | Client: hc [4]
13:20:02 2021-08-17 13:19:51:052 | Antistasi | Info | File: fn_loadNavGrid | Started loading nav grid | Client: hc [4]
13:20:02 2021-08-17 13:19:51:410 | Antistasi | Info | File: fn_loadNavGrid | Finished loading nav grid | Client: hc [4]
13:20:02 2021-08-17 13:19:51:410 | Antistasi | Info | File: initClient.sqf | Headless client version: 2.5.1 | Client: hc [4]
13:20:02 2021-08-17 13:19:51:410 | Antistasi | Info | File: initClient.sqf | Headless client Antistasi Plus version: 1.7 | Client: hc [4]
13:20:03 "Player/BIS_fnc_log: [BIS_fnc_preload] ----- Scripts initialized at 65298.8 ms -----"
13:20:36  Mission id: da71c2110963bd34c43e086c8f2082d4008d8e4f
13:20:36 Game started.
13:20:37 Strange convex component04 in a3\structures_f\ind\transmitter_tower\ttowerbig_2_ruins_f.p3d:geometryPhys
13:20:38 "6748.87: [Antistasi] | INFO | Headless Client Connected: [4]."
```

**Comment by igorkis-scrts** [comment]
I know the guy from Spain region too with the exact same error.
I may be wrong, but are connected to Gato "Ahmed" Pardo?

**Comment by Thadah** [comment]
No, sorry, I don't know the guy. We're just a bunch of friends on a private server.

**Comment by igorkis-scrts** [comment]
Okay, i asked because it was his own custom edit of Antistasi Plus and it was the only case at that moment.

**Comment by Thadah** [comment]
We didn't make any edit of Antistasi Plus.
Restoring a backup from yesterday has solved the issue.

**Comment by igorkis-scrts** [comment]
Could you execute this 
```
_markersX = markersX - watchpostsFIA - roadblocksFIA - aapostsFIA - atpostsFIA - mortarpostsFIA - hmgpostsFIA - controlsX;
copyToClipboard (str _markersX);
```
in debug console and post results there?

**Comment by Thadah** [comment]
"[""airport_1"",""airport_2"",""airport_3"",""airport_4"",""airport"",""resource"",""resource_1"",""resource_2"",""resource_4"",""resource_5"",""resource_6"",""resource_7"",""resource_3"",""factory"",""factory_1"",""factory_3"",""factory_4"",""outpost"",""outpost_1"",""outpost_10"",""outpost_11"",""outpost_12"",""outpost_13"",""outpost_14"",""outpost_2"",""outpost_3"",""outpost_4"",""outpost_5"",""outpost_6"",""outpost_7"",""outpost_8"",""outpost_9"",""outpost_15"",""outpost_16"",""seaport"",""seaport_1"",""seaport_2"",""seaport_3"",""seaport_4"",""seaport_5"",""milbase"",""milbase_1"",""Synd_HQ"",""Katkoula01"",""Yanukka01"",""Balavu01"",""Rautake01"",""Tavu01"",""Muaceba01"",""Namuvaka01"",""Laikoro01"",""Leqa01"",""IndPort01"",""Vatukoula01"",""Georgetown01"",""Lijnhaven01"",""Tuvanaka01"",""Belfort01"",""Koumac01"",""Moddergat01"",""Doodstil01"",""Blerick01"",""Harcourt01"",""PortBoise01"",""Tobakoro01"",""Losi01"",""Rereki01"",""Cerebu01"",""Savaka01"",""Lailai01"",""Tanouka01"",""Lifou01"",""LaRochelle01"",""Nicolet01"",""Regina01"",""Lami01"",""Kotomo01"",""Oumere01"",""Murarua01"",""Savu01"",""Buawa01"",""Ovau01"",""Vagalala01"",""Nasua01"",""Lobaka01"",""Nandai01"",""Momea01"",""Ba01"",""SaintJulien01"",""Imone01"",""PetitNicolet01"",""OuaOue01"",""Penelo01"",""SaintPaul01"",""Saioko01"",""Nani01"",""Buabua01"",""LaFoa01"",""Taga01""]"

This is on the working backup. I couldn't do it on the failed one.

**Comment by igorkis-scrts** [comment]
Could you also execute and post outcome on f this command?

```
_result = ["garrison"] call A3A_fnc_returnSavedStat;
copyToClipboard str _result;
```

**Comment by igorkis-scrts** [comment]
Execute this and try to load new save, it should fix it in a meantime.

```sqf
private _garrisonData = ["garrison"] call A3A_fnc_returnSavedStat;
{
    private _units = (_x#1) select {!isNil "_x"};
    _x set [1, _units];
} forEach _garrisonData;
["garrison",_garrisonData] call A3A_fnc_setStatVariable;
```

**Comment by Thadah** [comment]
It's ok. We have decided to use the backup in the end, as the loss has been negligible.

Still, many thanks for your quick and detailed support! I hope this ticket may help someone in the same situation.

**Comment by Thadah** [comment]
Hello! The bug happened again but Antistasi Plus 1.7.5 solved the issue, probably because of 348384ab274ee7fe9ede9131478e20e19040b314

Thanks for the bugfix!
